pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package upgrades
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/replication"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/upgrade"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+func maybeSetupPCRStandbyReader(
+	ctx context.Context, _ clusterversion.ClusterVersion, d upgrade.TenantDeps,
+) error {
+	if d.TenantInfoAccessor == nil {
+		return nil
+	}
+	id, ts, err := d.TenantInfoAccessor.ReadFromTenantInfo(ctx)
+	if err != nil {
+		return err
+	}
+	if !id.IsSet() {
+		return nil
+	}
+	if ts.IsEmpty() {
+		log.Warning(ctx, "replicated timestamp is zero")
```

Comment:
`return nil` ?

Reply:
we need to set up the poller job even if the timestamp is 0

Reply:
if the ts is 0, we just need to avoid calling `SetupOrAdvanceStandbyReaderCatalog`

Reply:
But only creating the poller works too and is arguably the more defensive approach

Reply:
I updated the code to create the poller job only if replicated time is zero